File: Ruby/192bit/Rijandel.rb
require 'openssl'
require 'securerandom'

def aes_encrypt(plaintext, key)
  cipher = OpenSSL::Cipher.new('aes-192-cbc')
  cipher.encrypt
  iv = cipher.random_iv
  cipher.key = key

  ciphertext = cipher.update(plaintext) + cipher.final
  iv + ciphertext
end

def main
  key = SecureRandom.random_bytes(24) # 192-bit key
  plaintext = "This is a test message for AES-192 encryption!"

  ciphertext = aes_encrypt(plaintext, key)

  puts "Ciphertext is: #{ciphertext.unpack1('H*')}"
end

main

File: Python/192bit/Rijandel.py
from Crypto.Cipher import AES
from Crypto.Random import get_random_bytes

def aes_encrypt(plaintext, key):
    cipher = AES.new(key, AES.MODE_CBC)
    ciphertext = cipher.encrypt(plaintext)
    return cipher.iv + ciphertext

def pad(data):
    pad_len = AES.block_size - len(data) % AES.block_size
    return data + bytes([pad_len] * pad_len)

def main():
    key = get_random_bytes(24)  # 192-bit key
    plaintext = b"This is a test message for AES-192 encryption!"
    plaintext = pad(plaintext)

    ciphertext = aes_encrypt(plaintext, key)

    print("Ciphertext is:", ciphertext.hex())

if __name__ == "__main__":
    main()
